relay-cardinality/src/limiter.rs

```diff
@@ -172,14 +215,16 @@ pub struct CardinalityLimits<'a, T> {
     source: Vec<T>,
     rejections: HashSet<usize>,
     limits: HashSet<&'a CardinalityLimit>,
+    reports: BTreeMap<&'a CardinalityLimit, Vec<CardinalityReport>>,
 }
 
 impl<'a, T> CardinalityLimits<'a, T> {
-    fn new(source: Vec<T>, rejections: RejectionTracker<'a>) -> Self {
+    fn new(source: Vec<T>, reporter: DefaultReporter<'a>) -> Self {
         Self {
             source,
-            rejections: rejections.entries,
-            limits: rejections.limits,
+            rejections: reporter.entries,
+            limits: reporter.limits,
+            reports: reporter.reports,
```

Comment:
Should we make `reporter` an attribute of `CardinalityLimits`, instead of tracking these 3 separately?

Reply:
Mh, I tried it just now, but it makes a bunch of lines break because the variables are now longer.